app/controllers/public/index.js

```diff
@@ -10,6 +11,19 @@ export default Controller.extend({
   { name: 'Speaker 4', organisation: 'Org 4', isFeatured: true, socialLinks: [{ name: 'twitter', url: '#' }], short_biography: 'Full-stack developer with experience in PHP, Python &amp; Ember.JS. Plays around with Kubernetes and Containers. Avid open source contributor.' },
   { name: 'Speaker 5', organisation: 'Org 5', isFeatured: false, socialLinks: [{ name: 'home', url: '#' }],  short_biography: 'Full-stack developer with experience in PHP, Python &amp; Ember.JS. Plays around with Kubernetes and Containers. Avid open source contributor.' }],
 
+  sponsors: [{ name: 'Sponsor 2', url: '#', logoUrl: 'http://placehold.it/150x60', level: 2, type: 'Gold Sponsor', description: 'Blah Blah Blah' },
+    { name: 'Sponsor 1', url: '#', logoUrl: 'http://placehold.it/150x60', level: 1, type: 'Gold Sponsor', description: 'Blah Blah Blah' },
+    { name: 'Sponsor 3', url: '#', logoUrl: 'http://placehold.it/150x60', level: 1, type: 'Silver Sponsor', description: 'Blah Blah Blah' }
+  ],
+
+  sponsorsByCategory: computed(('sponsors'), function() {
```

Comment:
@geekyd 

This is too complex. We're doing needles operations here. this can be simplified to just a single sortBy and a groupBy. In a javascript single page application like this, we need to take atmost care to ensure we use as less memory and resources as possible, since it all happens on the client side and we can't always expect our clients to have good PCs.

See this alternative implementation,

```js
sponsorsByCategory: computed('sponsors', function() {
    return _.groupBy(_.orderBy(this.get('sponsors'), 'level'), 'type');;
})
```

This will now return an object in the form 

```js
{
  "Gold Sponsor": [
    {
      "name": "Sponsor 1",
      "url": "#",
      "logoUrl": "http://placehold.it/150x60",
      "level": 1,
      "type": "Gold Sponsor",
      "description": "Blah Blah Blah"
    },
    {
      "name": "Sponsor 2",
      "url": "#",
      "logoUrl": "http://placehold.it/150x60",
      "level": 2,
      "type": "Gold Sponsor",
      "description": "Blah Blah Blah"
    }
  ],
  "Silver Sponsor": [
    {
      "name": "Sponsor 3",
      "url": "#",
      "logoUrl": "http://placehold.it/150x60",
      "level": 1,
      "type": "Silver Sponsor",
      "description": "Blah Blah Blah"
    }
  ]
}
```

Now, in the template, You can use [`{{#each-in object as |key value|}}  {{/each-in}}`](https://www.emberjs.com/api/classes/Ember.Templates.helpers.html#method_each-in) instead of `each` to loop through this.